lua/rainbow-delimiters/strategy/global.lua

```diff
@@ -64,36 +78,63 @@ local function update_range(bufnr, changes, tree, lang)
 	local matches = Stack.new()
 
 	for _, change in ipairs(changes) do
+		-- This is the match record, it lists all the relevant nodes from
+		-- each match.
+		local match_record = new_match_record()
 		local root_node = tree:root()
 		local start_row, end_row = change[1], change[3] + 1
 		lib.clear_namespace(bufnr, lang, start_row, end_row)
-		for _, match, _ in query:iter_matches(root_node, bufnr, start_row, end_row) do
-			-- This is the match record, it lists all the relevant nodes from
-			-- the match.
-			local match_record = {
-				opening = Stack.new(),
-				closing = Stack.new(),
-				intermediate = Stack.new(),
-				children = Stack.new(),
-			}
-			for id, node in pairs(match) do
-				local name = query.captures[id]
-				if name == 'container' then
-					match_record.container = node
-				else
-					if match_record[name] then match_record[name]:push(node) end
+		for qid, node, _ in query:iter_captures(root_node, bufnr, start_row, end_row) do
+			local name = query.captures[qid]
+			-- check for 'delimiter' first, since that should be the most
+			-- common name
+			if name == 'delimiter' then
+				match_record.delimiter:push(node)
+			elseif name == 'container' and not match_record.container then
+				match_record.container = node
```

Comment:
With the below change this can be changed to
```suggestion
			elseif name == 'container' and not match_record then
				match_record = new_match_record(node)
```